### PR TITLE
alternate workflow for "windows-ci" to accomodate documentation-only PRs.

### PR DESCRIPTION
**Pull request by TimothyMothra**
#3106.

## Changes
- new workflow "windows-ci-md" for documentation-only PRs. same name as "windows-ci"

Please provide a brief description of the changes here.

I tested a few iterations of this on my forked repo.
The `name` and matrix values are required to match.

For normal PRs (not documentation),
I tried excluding this workflow but wasn't successful.
I verified it's not detrimental to have this run in addition to required workflows.
However it does make the list of Checks a little bit confusing with duplicate names.

https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks

> If someone submits a pull request that changes a markdown file in the root of the repository, then the workflow above won't run at all because of the path filtering. As a result you won't be able to merge the pull request. You would see the following status on the pull request:
You can fix this by creating a generic workflow, with the same name, that will return true in any case similar to the workflow below :

For significant contributions please make sure you have completed the following items:

* [ ] Appropriate `CHANGELOG.md` updated for non-trivial changes
* [ ] Design discussion issue #
* [ ] Changes in public API reviewed

### CODE REVIEW

**Comment by TimothyMothra** [comment]
I tested this in my forked repo and confirmed that the negative filter `!**.md`

**Comment by TimothyMothra** [comment]
> Thanks. Will try this and let you know

This may need something extra to work with a version matrix. 
I'm testing in my fork.

I don't seem to have permission to change this PR to a DRAFT

File: .github/workflows/windows-ci-md.yml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    paths-ignore:
+    - '!**.md'
+
+jobs:
+  build-test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        version: [net461,netcoreapp3.1,net5.0,net6.0]
```
Comment (reyang) [inline .github/workflows/windows-ci-md.yml:22]: Is this line needed?

**Review by reyang — APPROVED** [review]

File: .github/workflows/windows-ci-md.yml (hunk 1 above)
Comment (TimothyMothra) [reply .github/workflows/windows-ci-md.yml:22]: It appears to be required, yes.

The Required check includes the value from the matrix.
When testing without, that check isn't run.

Comment (reyang) [reply .github/workflows/windows-ci-md.yml:22]: Thanks for confirmation!

**Pull request merged**